### PR TITLE
Ensure that the top scroll bar responds to column and data changes

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
#104

### CODE REVIEW

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://developer.blackbaud.com/skyux-grids?_sp.spa=pr-top-scroll-changes-detection](https://developer.blackbaud.com/skyux-grids?_sp.spa=pr-top-scroll-changes-detection)

File: src/app/public/modules/grid/grid.component.ts [hunk 1]
```diff
@@ -750,6 +752,10 @@ export class SkyGridComponent implements OnInit, AfterContentInit, AfterViewInit
     this.getColumnsFromComponent();
     this.setDisplayedColumns(true);
     this.ref.markForCheck();
+
+    setTimeout(() => {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/grid/grid.component.ts]: I notice we're calling `checkUserColumnWidthsForScroll` inside a `setTimeout` in certain places, but not others. For example, in the `ngAfterViewInit`. Any rhyme or reason to this?

Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/grid/grid.component.ts]: Yup. When the data or the columns change things may populate into the typescript prior to the actual window updating. The set timeouts are necessary to ensure that the view is updated before we try to measure widths. In the other cases things are rendered already when we do the checking.

I added a couple comments to clarify this

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://developer.blackbaud.com/skyux-grids?_sp.spa=pr-top-scroll-changes-detection](https://developer.blackbaud.com/skyux-grids?_sp.spa=pr-top-scroll-changes-detection)

File: src/app/public/modules/grid/grid.component.ts (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.ts]: Thanks. Sounds good 👍

**Pull request merged**